lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,184 @@
+  const Sample outputSample(functionalChaosResult_.getOutputSample());
+  const Function metamodel(functionalChaosResult_.getMetaModel());
+  const Sample outputMetamodel(metamodel(inputSample));
+  const UnsignedInteger outputDimension = outputSample.getDimension();
+  const UnsignedInteger sampleSize = inputSample.getSize();
+  Point leaveOneOutMSE(outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+  {
+    const Sample marginalOutputSample(outputSample.getMarginal(j));
+    const Point marginalOutputPoint(marginalOutputSample.asPoint());
+    // Necessary to compute the projection matrix
+    const Point betaHat = leastSquaresMethod.solve(marginalOutputPoint);
+    const Point diagonalH = leastSquaresMethod.getHDiag();
+    const Sample outputMetamodelMarginal(outputMetamodel.getMarginal(j));
+    const Point residuals = marginalOutputPoint - outputMetamodelMarginal.asPoint();
+    Point correctedResiduals(sampleSize);
+    Point squaredCorrectedResiduals(sampleSize);
+    for (UnsignedInteger i = 0; i < sampleSize; ++i)
+    {
+      correctedResiduals[i] = residuals[i] / (1.0 - diagonalH[i]);
+      squaredCorrectedResiduals[i] = correctedResiduals[i] * correctedResiduals[i];
+    }
+    const Sample squaredCorrectedResidualsSample(Sample::BuildFromPoint(squaredCorrectedResiduals));
+    leaveOneOutMSE[j] = squaredCorrectedResidualsSample.computeMean()[0];
+  } // For output marginal indices
+  return leaveOneOutMSE;
+}
+
+/* Compute K-Fold error */
+Point FunctionalChaosValidation::computeKFoldError(const UnsignedInteger & k) const
+{
+  const Sample inputSample(functionalChaosResult_.getInputSample());
+  const UnsignedInteger sampleSize = inputSample.getSize();
+  const Sample outputSample(functionalChaosResult_.getOutputSample());
+  const UnsignedInteger outputDimension = outputSample.getDimension();
+  const Function metamodel(functionalChaosResult_.getMetaModel());
+  const Function transformation(functionalChaosResult_.getTransformation());
+  const Sample standardSample(transformation(inputSample));
+  const FunctionCollection reducedBasis(functionalChaosResult_.getReducedBasis());
+  DesignProxy designProxy(standardSample, reducedBasis);
+  const UnsignedInteger reducedBasisSize = reducedBasis.getSize();
+  Indices allIndices(reducedBasisSize);
+  allIndices.fill();
+  // The method name is set to the default one, given by ResourceMap
+  String methodName = ResourceMap::GetAsString("LeastSquaresExpansion-DecompositionMethod");
+  LeastSquaresMethod leastSquaresMethod = LeastSquaresMethod::Build(methodName, designProxy, allIndices);
+  Point kFoldMSE(outputDimension);
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+  {
+    const Sample marginalOutputSample(outputSample.getMarginal(j));
+    const Point marginalOutputPoint(marginalOutputSample.asPoint());
+    // Necessary to compute the projection matrix
+    const Point betaHat = leastSquaresMethod.solve(marginalOutputPoint);
```

Comment:
Don't use affectation here

Reply:
I removed this code. This must be replace by a call to `update()`, that I did not know of. This topic can be considered as "resolved by collateral damage".